Dockerfile

```diff
@@ -1,10 +1,12 @@
-FROM ubuntu:20.04
-RUN apt-get update && apt-get upgrade --yes && apt-get install git python3.8 python3-pip --yes
+FROM ubuntu:22.04
+RUN apt-get update && apt-get upgrade --yes && apt-get install git python3.9 python3-pip --yes
 ADD . /code
 WORKDIR /code
 
 # need to use typing-extensions<4 to deal with issue 387 bug: Docker will not build
 # TODO: Figure out a better solution or wait for it to resolve itself.
 RUN pip install typing-extensions==3.10.0.2
-RUN pip install .[recommended-plugins]
+RUN pip install .
+RUN ape plugins install alchemy ens etherscan foundry hardhat infura ledger solidity template tokens trezor vyper
```

Comment:
no it doesn't. If I revert there is an error that it can't install the plugins properly. I will update this comment shortly with the error.

Reply:
@fubuloubu Things tend to work better across the board when installing separately versus using `setup.py` keys

Reply:
` => => # INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce  => => # runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C. `
and 
```
#11 124.5 The conflict is caused by:
#11 124.5     eth-ape[recommended-plugins] 0.3.6.dev13+g5d1daf9f.d20220719 depends on eth-ape 0.3.6.dev13+g5d1daf9f.d20220719 (from /code)
#11 124.5     ape-alchemy 0.2.0 depends on eth-ape<0.3.0 and >=0.2.1
#11 124.5     eth-ape[recommended-plugins] 0.3.6.dev13+g5d1daf9f.d20220719 depends on eth-ape 0.3.6.dev13+g5d1daf9f.d20220719 (from /code)
#11 124.5     ape-alchemy 0.1.1 depends on eth-ape<0.3.0 and >=0.2.1
#11 124.5     eth-ape[recommended-plugins] 0.3.6.dev13+g5d1daf9f.d20220719 depends on eth-ape 0.3.6.dev13+g5d1daf9f.d20220719 (from /code)
#11 124.5     ape-alchemy 0.1.0 depends on eth-ape<0.2.0 and >=0.1.0
#11 124.5 
#11 124.5 To fix this you could try to:
#11 124.5 1. loosen the range of package versions you've specified
#11 124.5 2. remove package versions to allow pip attempt to solve the dependency conflict
#11 124.5 
#11 124.5 ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
------
executor failed running [/bin/sh -c pip install .[recommended-plugins]]: exit code: 1
```

Reply:
The plugins needs eth-ape and eth-ape needs the plugins, the cyclic dependency causes pip to have to work really hard (in the best case scenario that is succeeds)

Reply:
Ah, this seems like a nasty bug!

Reply:
> Ah, this seems like a nasty bug!

I think it might actually be two bugs because the `INFO` one with the cyclic dependencies happens when I do that command in regular ape (non-docker) and although it doesn't seem to fail it takes forever to install.